plugins/org.locationtech.udig.catalog.oracle/src/org/locationtech/udig/catalog/internal/oracle/OraclePlugin.java

```diff
@@ -23,115 +23,130 @@
  * The main plugin class to be used in the desktop.
  */
 public class OraclePlugin extends AbstractUIPlugin {
-	//The shared instance.
-	private static OraclePlugin plugin;
-	//Resource bundle.
-	private ResourceBundle resourceBundle;
-	
+
+    private static OraclePlugin plugin = new OraclePlugin();
+
+    private ResourceBundle resourceBundle;
+
     public static final String ID = "org.locationtech.udig.catalog.oracle"; //$NON-NLS-1$
-	/**
-	 * The constructor.
-	 */
-	public OraclePlugin() {
-		super();
-		plugin = this;
-	}
 
-	/**
-	 * This method is called upon plug-in activation
-	 */
-	public void start(BundleContext context) throws Exception {
-		super.start(context);
-	}
+    /**
+     * The constructor.
+     */
+    private OraclePlugin() {
+        super();
+        plugin = this;
+    }
+
+    /**
+     * This method is called upon plug-in activation
+     */
+    @Override
+    public void start(BundleContext context) throws Exception {
+        super.start(context);
+    }
+
+    /**
+     * This method is called when the plug-in is stopped
+     */
+    @Override
+    public void stop(BundleContext context) throws Exception {
+        plugin = null;
+        resourceBundle = null;
+        super.stop(context);
+    }
```

Comment:
Hmm, wondering a bit since Eclipse generates Activator-class with:
```
	// The shared instance
	private static Activator plugin;
	
	/**
	 * The constructor
	 */
	public Activator() {
	}

	@Override
	public void start(BundleContext context) throws Exception {
		super.start(context);
		plugin = this;
	}

	@Override
	public void stop(BundleContext context) throws Exception {
		plugin = null;
		super.stop(context);
	}

	/**
	 * Returns the shared instance
	 *
	 * @return the shared instance
	 */
	public static Activator getDefault() {
		return plugin;
	}
```

I've checked such a class locally with Eclipse Spotbugs Integration but without any complains. @sschulz92 What kind of error/warning have you fixed with this change?